wpcom-legacy-redirector.php

```diff
@@ -104,13 +105,58 @@ static function insert_legacy_redirect( $from_url, $redirect_to ) {
 			return new WP_Error( 'invalid-redirect-url', 'Invalid redirect_to param; should be a post_id or a URL' );
 		}
 
-		wp_insert_post( $args );
+		switch ( $mode ) {
+			case 'update':
+				$args[ 'ID' ] = self::get_redirect_post_id( $from_url, false );
+				if ( 0 === wp_update_post( $args ) ) {
+					return new WP_Error( 'failed-redirect-update', 'Redirect update failed for unknown reason' );
+				}
+				break;
+			case 'add':
+			default:
+				wp_insert_post( $args );
```

Comment:
similarly to how the code is checking whether the `wp_update_post` succeeded, we perhaps should check the return value of the `wp_insert_post` action before returning true.

I know that it hasn't been doing that prior this update, so feel free to leave that for a separate enhancement.